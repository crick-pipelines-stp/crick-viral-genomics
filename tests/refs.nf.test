nextflow_pipeline {
    name "Test Pipeline Components"
    script "../main.nf"

    tag "refs"

    test("refs_host_genome_resolve_illumina") {
        tag "refs/host_genome_resolve_illumina"
        when {
            params {
                run_remove_host_reads = true
                run_reporting         = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp_min.csv"
                host_genome = "sacCer3"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("refs_host_genome_resolve_nanopore") {
        tag "refs/host_genome_resolve_nanopore"
        when {
            params {
                run_remove_host_reads = true
                run_minimap_align     = true
                run_reporting         = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp_min.csv"
                host_genome = "sacCer3"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("refs_merge_refs") {
        tag "refs/merge_refs"
        when {
            params {
                run_reporting = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_flu_small.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }
}
