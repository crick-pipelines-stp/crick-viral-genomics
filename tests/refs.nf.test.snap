{
    "refs_host_genome_resolve_nanopore": {
        "content": [
            42,
            [
                "alignment",
                "alignment/FAY66992_BC15.bam",
                "alignment/FAY66992_BC15.viral.sorted.bam",
                "alignment/FAY66992_BC15.viral.sorted.bam.bai",
                "alignment/FAY66992_BC23.bam",
                "alignment/FAY66992_BC23.viral.sorted.bam",
                "alignment/FAY66992_BC23.viral.sorted.bam.bai",
                "alignment/stats",
                "alignment/stats/FAY66992_BC15.viral.flagstat",
                "alignment/stats/FAY66992_BC15.viral.idxstats",
                "alignment/stats/FAY66992_BC15.viral.stats",
                "alignment/stats/FAY66992_BC23.viral.flagstat",
                "alignment/stats/FAY66992_BC23.viral.idxstats",
                "alignment/stats/FAY66992_BC23.viral.stats",
                "consensus",
                "consensus/consensus.merged.fasta",
                "consensus/count_tables",
                "consensus/count_tables/FAY66992_BC15.count_table.csv",
                "consensus/count_tables/FAY66992_BC23.count_table.csv",
                "consensus/pileup",
                "consensus/pileup/FAY66992_BC15.mpileup",
                "consensus/pileup/FAY66992_BC23.mpileup",
                "host",
                "host/FAY66992_BC15.host.sorted.bam",
                "host/FAY66992_BC15.host.sorted.bam.bai",
                "host/FAY66992_BC23.host.sorted.bam",
                "host/FAY66992_BC23.host.sorted.bam.bai",
                "host/stats",
                "host/stats/FAY66992_BC15.host.flagstat",
                "host/stats/FAY66992_BC15.host.idxstats",
                "host/stats/FAY66992_BC15.host.stats",
                "host/stats/FAY66992_BC23.host.flagstat",
                "host/stats/FAY66992_BC23.host.idxstats",
                "host/stats/FAY66992_BC23.host.stats",
                "pipeline_info",
                "qc",
                "qc/mosdepth",
                "qc/mosdepth/FAY66992_BC15.mosdepth.global.dist.txt",
                "qc/mosdepth/FAY66992_BC15.mosdepth.region.dist.txt",
                "qc/mosdepth/FAY66992_BC15.mosdepth.summary.txt",
                "qc/mosdepth/FAY66992_BC15.per-base.bed.gz",
                "qc/mosdepth/FAY66992_BC15.per-base.bed.gz.csi",
                "qc/mosdepth/FAY66992_BC15.regions.bed.gz",
                "qc/mosdepth/FAY66992_BC15.regions.bed.gz.csi",
                "qc/mosdepth/FAY66992_BC23.mosdepth.global.dist.txt",
                "qc/mosdepth/FAY66992_BC23.mosdepth.region.dist.txt",
                "qc/mosdepth/FAY66992_BC23.mosdepth.summary.txt",
                "qc/mosdepth/FAY66992_BC23.per-base.bed.gz",
                "qc/mosdepth/FAY66992_BC23.per-base.bed.gz.csi",
                "qc/mosdepth/FAY66992_BC23.regions.bed.gz",
                "qc/mosdepth/FAY66992_BC23.regions.bed.gz.csi",
                "ref_assembly",
                "ref_assembly/SARS-CoV-2.fasta.fai",
                "viral_fastq",
                "viral_fastq/FAY66992_BC15.viral_1.fastq.gz",
                "viral_fastq/FAY66992_BC23.viral_1.fastq.gz"
            ],
            [
                "FAY66992_BC15.bam:md5,653a21851652760deff62cfcc92c936a",
                "FAY66992_BC15.viral.sorted.bam:md5,399ae9b542547eb4f56a18e95864e474",
                "FAY66992_BC15.viral.sorted.bam.bai:md5,fde097e2bc3ba27d103a542c9557ae8c",
                "FAY66992_BC23.bam:md5,0624256f33ca7baa4fb7c1955c07f4ca",
                "FAY66992_BC23.viral.sorted.bam:md5,3db0ed052b286d2cccda5ab474aa74ed",
                "FAY66992_BC23.viral.sorted.bam.bai:md5,cc9c2434fe2b791a9ec4b7a49f857e84",
                "FAY66992_BC15.viral.flagstat:md5,717d75014cc36984b4c609cf88ceb492",
                "FAY66992_BC15.viral.idxstats:md5,b4e145beedadd522f1fd2cd1f95d9b6a",
                "FAY66992_BC15.viral.stats:md5,3b828748a3d82bbcafe1b7351c1b0403",
                "FAY66992_BC23.viral.flagstat:md5,49baf7593bbd9a3f936b146a36800caf",
                "FAY66992_BC23.viral.idxstats:md5,66f80fe4ec6d20c555c34972426776ac",
                "FAY66992_BC23.viral.stats:md5,47a905bd28f6a710118b1180d92b3e9b",
                "consensus.merged.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC15.count_table.csv:md5,cc30c9f4c332c4e2442f4e414f1e0de3",
                "FAY66992_BC23.count_table.csv:md5,0f8e01f23205854196483e77362aa8c3",
                "FAY66992_BC15.mpileup:md5,4503b6a1d80eeaf2ded2978f2c5abce4",
                "FAY66992_BC23.mpileup:md5,79e15fe801c5dd33c2fc4b1839acf2e0",
                "FAY66992_BC15.host.sorted.bam:md5,c35dfbc1d11ac38e8eec83d4aeee63de",
                "FAY66992_BC15.host.sorted.bam.bai:md5,e91dce50260434b47f311e6867b12bcf",
                "FAY66992_BC23.host.sorted.bam:md5,8d9f797030c877ca85144463003c54c9",
                "FAY66992_BC23.host.sorted.bam.bai:md5,b500b9e5bbbfa3d1f00391c1d2049d82",
                "FAY66992_BC15.host.flagstat:md5,e08521543a095801f48f59c6884aa135",
                "FAY66992_BC15.host.idxstats:md5,646c153849e6691daa9e1a81e13dddc0",
                "FAY66992_BC15.host.stats:md5,9f477847d976dd870d67d362a4dfa02e",
                "FAY66992_BC23.host.flagstat:md5,e2060f460793696320d456702d2b3ef4",
                "FAY66992_BC23.host.idxstats:md5,c66dbaa2850b0c0d889befe292b5db20",
                "FAY66992_BC23.host.stats:md5,7a9f72a103f004d551de931b0afb6032",
                "FAY66992_BC15.mosdepth.global.dist.txt:md5,d97850e4a9407d08fb700330078e1225",
                "FAY66992_BC15.mosdepth.region.dist.txt:md5,9b0b039931c43acb2b7f34b3760fb16c",
                "FAY66992_BC15.mosdepth.summary.txt:md5,516d5e42bf42a9a714fe48d57fa8b3fc",
                "FAY66992_BC15.per-base.bed.gz:md5,395d05ccad354cb580745853a8a52ae7",
                "FAY66992_BC15.per-base.bed.gz.csi:md5,231ff388a77696429c3e0b4a89906307",
                "FAY66992_BC15.regions.bed.gz:md5,0282d47e59965a4cb7c33f1de91a08f3",
                "FAY66992_BC15.regions.bed.gz.csi:md5,a5ddee8992b31a701a23a9ac4d02d310",
                "FAY66992_BC23.mosdepth.global.dist.txt:md5,add230bdb21ec5045d5f45b5b03fcb34",
                "FAY66992_BC23.mosdepth.region.dist.txt:md5,50aae8fd6e67d6434dc96cd99bb2a9bc",
                "FAY66992_BC23.mosdepth.summary.txt:md5,506d22517700faa4324a25502d08f6f7",
                "FAY66992_BC23.per-base.bed.gz:md5,65e9ab7d62122c132a352b1a7cc3907d",
                "FAY66992_BC23.per-base.bed.gz.csi:md5,6fa47b28b61c2fd8d93dc7e1400c8c1e",
                "FAY66992_BC23.regions.bed.gz:md5,b5f762820e63a0dff0590266db1f10f9",
                "FAY66992_BC23.regions.bed.gz.csi:md5,a5ddee8992b31a701a23a9ac4d02d310",
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2",
                "FAY66992_BC15.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC23.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-07T09:10:11.480213"
    },
    "refs_force_upper": {
        "content": [
            6,
            [
                "consensus",
                "consensus/consensus.merged.fasta",
                "pipeline_info",
                "ref_assembly",
                "ref_assembly/viral_reference.upper.fasta.fai"
            ],
            [
                "consensus.merged.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                "viral_reference.upper.fasta.fai:md5,c1207acbd8f4d0ce7d301a735c40b378"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-07T10:08:27.106143"
    },
    "refs_host_genome_resolve_illumina": {
        "content": [
            22,
            [
                "consensus",
                "consensus/consensus.merged.fasta",
                "host",
                "host/FAY66992_BC15.host.sorted.bam",
                "host/FAY66992_BC15.host.sorted.bam.bai",
                "host/FAY66992_BC23.host.sorted.bam",
                "host/FAY66992_BC23.host.sorted.bam.bai",
                "host/stats",
                "host/stats/FAY66992_BC15.host.flagstat",
                "host/stats/FAY66992_BC15.host.idxstats",
                "host/stats/FAY66992_BC15.host.stats",
                "host/stats/FAY66992_BC23.host.flagstat",
                "host/stats/FAY66992_BC23.host.idxstats",
                "host/stats/FAY66992_BC23.host.stats",
                "pipeline_info",
                "ref_assembly",
                "ref_assembly/SARS-CoV-2.fasta.fai",
                "viral_fastq",
                "viral_fastq/FAY66992_BC15.viral_1.fastq.gz",
                "viral_fastq/FAY66992_BC23.viral_1.fastq.gz"
            ],
            [
                "consensus.merged.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC15.host.sorted.bam:md5,afe07b831e6fb499e3f9e16e83e7f6ec",
                "FAY66992_BC15.host.sorted.bam.bai:md5,83753457460d39f364618022763a229f",
                "FAY66992_BC23.host.sorted.bam:md5,383c6f2a6c9a7ea73f20d5f7a3710fab",
                "FAY66992_BC23.host.sorted.bam.bai:md5,b500b9e5bbbfa3d1f00391c1d2049d82",
                "FAY66992_BC15.host.flagstat:md5,97c4d446d23e33217115d33debfdd7fb",
                "FAY66992_BC15.host.idxstats:md5,e1c6d291c21fd6a4e3ebf2e0b56bdeba",
                "FAY66992_BC15.host.stats:md5,d8672e15b5110c7722bbb854e2067a73",
                "FAY66992_BC23.host.flagstat:md5,e2060f460793696320d456702d2b3ef4",
                "FAY66992_BC23.host.idxstats:md5,c66dbaa2850b0c0d889befe292b5db20",
                "FAY66992_BC23.host.stats:md5,7a9f72a103f004d551de931b0afb6032",
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2",
                "FAY66992_BC15.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "FAY66992_BC23.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-07T09:09:46.878718"
    },
    "refs_merge_refs": {
        "content": [
            5,
            [
                "consensus",
                "consensus/consensus.merged.fasta",
                "pipeline_info",
                "ref_assembly",
                "ref_assembly/viral_reference.merged.fasta.fai"
            ],
            [
                "consensus.merged.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                "viral_reference.merged.fasta.fai:md5,c1207acbd8f4d0ce7d301a735c40b378"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-07T10:05:47.621749"
    }
}